module/input/SensorFilter/src/SensorFilter.cpp

```diff
@@ -209,20 +238,28 @@ namespace module::input {
                         // Make sure we have servo positions
                         for (uint32_t id = 0; id < 20; ++id) {
                             auto& original = utility::platform::getRawServo(id, *s);
-                            // Add the sensor values to the system properly
-                            filtered_sensors.servo.push_back({0,
-                                                              id,
-                                                              original.torque_enabled,
-                                                              original.p_gain,
-                                                              original.i_gain,
-                                                              original.d_gain,
-                                                              original.goal_position,
-                                                              original.moving_speed,
-                                                              original.present_position,
-                                                              original.present_speed,
-                                                              original.load,
-                                                              original.voltage,
-                                                              static_cast<float>(original.temperature)});
+                            auto& error    = original.error_flags;
+
+                            // Check for an error on the servo and report it
+                            if (error != RawSensors::Error::OK) {
+                                NUClear::log<NUClear::WARN>(servoErrorString(id, original, error));
+                            }
+                            else {
+                                // Add the sensor values to the system properly
+                                filtered_sensors.servo.push_back({0,
+                                                                  id,
+                                                                  original.torque_enabled,
+                                                                  original.p_gain,
+                                                                  original.i_gain,
+                                                                  original.d_gain,
+                                                                  original.goal_position,
+                                                                  original.moving_speed,
+                                                                  original.present_position,
+                                                                  original.present_speed,
+                                                                  original.load,
+                                                                  original.voltage,
+                                                                  static_cast<float>(original.temperature)});
+                            }
```

Comment:
I think this might have an issue - we accumulate then divide by the size. If we have errors on the servos, we are dividing by a number larger than the number of things accumulated, and so it won't be the average.

Reply:
Ah, I was going to change that but forgot, thanks

Reply:
Actually no, i don't think we use the amount of servos we have to calculate averages we only use the number of sensor messages we have